### ISSUE TITLE
There should be some warning before deleting the default Execution Environment

### ISSUE THREAD

**Issue by AlanCoding** [ansible/awx#15303]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Feature type

Enhancement to Existing Feature

### Feature Summary

Give some notice before the default execution environment gets deleted.

(all steps as superuser)

### Select the relevant components

- [X] UI
- [X] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [ ] Other

### Steps to reproduce

If I got to the UI page for the default EE, it looks like this:

http://localhost:8013/#/execution_environments/1/details

Then I can just click "Delete", and it deletes as if it is an ordinary Execution Environment.

### Current results

Then all jobs will give this error:

![Screenshot from 2024-06-26 08-43-08](https://github.com/ansible/awx/assets/1385596/ec59d831-5a78-4325-9016-7025204d73c0)

...assuming the job doesn't have a custom EE set. Which would match the Demo Job Template.

### Sugested feature result

Somewhere in the process of clicking "Delete" it denies the request or gives me a warning that says specifically that it is the default EE in the system.

I would argue that an error is okay, because the user should, ideally, change the setting to a _different_ EE, then come back and delete the old one.

**Comment by marek1712** [comment]
Interesting. Delete is greyed out for **Control Plane Execution Environment**.
Furthermore, I get this when I try to delete any other EE:
![obraz](https://github.com/ansible/awx/assets/107187302/21acd8db-5504-4245-bebd-7728f1607429)
I'm running 24.6.0.

**Comment by AlanCoding** [comment]
Yeah, but it's a generic message. There's no indicator making it clear that particular image is the default. I honestly don't know for sure that it _is_ the default. You could have kept around some old versions of AWX EE.
